src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs

```diff
@@ -1234,7 +1227,66 @@ internal SourceFile(AssemblyInfo assembly, int id, DocumentHandle docHandle, Uri
             string escapedDocumentName = EscapePathForUri(documentName.Replace("\\", "/"));
             this.FileUriEscaped = $"file://{(OperatingSystem.IsWindows() ? "/" : "")}{escapedDocumentName}";
             this.DotNetUrlEscaped = $"dotnet://{assembly.Name}/{escapedDocumentName}";
-            this.Url = new Uri(File.Exists(documentName) ? FileUriEscaped : DotNetUrlEscaped, UriKind.Absolute);
+            if (!File.Exists(documentName) && SourceLinkUri != null)
+            {
+                string sourceLinkCachedPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "SourceServer", GetHashOfString(SourceLinkUri.AbsoluteUri), relativePath);
```

Comment:
nit: move to a local, since we are using it in two places and it would avoid them getting out of sync.

Reply:
`Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "SourceServer", GetHashOfString(SourceLinkUri.AbsoluteUri)`